src/sql-parser/src/parser.rs

```diff
@@ -1659,9 +1659,50 @@ impl<'a> Parser<'a> {
             envelope,
             if_not_exists,
             materialized,
+            key_constraint,
         }))
     }
 
+    /// Parses the column section of a CREATE SOURCE statement which can be
+    /// empty, a comma-separated list of column identifier or key constraint,
+    /// or a single key constraint.
+    fn parse_source_columns(&mut self) -> Result<(Vec<Ident>, Option<KeyConstraint>), ParserError> {
+        if self.consume_token(&Token::LParen) {
+            let mut columns = vec![];
+            let mut key_constraints = vec![];
+            loop {
+                let pos = self.peek_pos();
+                if let Some(key_constraint) = self.parse_key_constraint()? {
+                    if !key_constraints.is_empty() {
+                        return parser_err!(self, pos, "Multiple key constraints not allowed");
+                    }
+                    key_constraints.push(key_constraint);
+                } else {
+                    columns.push(self.parse_identifier()?);
+                }
+                if !self.consume_token(&Token::Comma) {
+                    break;
+                }
+            }
+            self.expect_token(&Token::RParen)?;
+            Ok((columns, key_constraints.into_iter().next()))
+        } else {
+            Ok((vec![], self.parse_key_constraint()?))
```

Comment:
Strikes me as odd that key constraints can occur outside parentheses because PG's constraints all must be parenthesized. I cede that this syntax is different, but in my mind sources are kind of an analog to `CREATE TABLE AS`...though that statement doesn't support constraints whatsoever. ¯\\\_(ツ)\_/¯

Could we err on the side of being more restrictive (i.e. requiring parentheses) and allowing unparenthesized constraints at some point in the future?

I don't have a compelling reason why this should be required other than cargo-culting PG, so if there was much thought put into the consideration, definitely not a hard requirement.

Reply:
I also have `CREATE SOURCE` and `CREATE TABLE` close to each other mentally, so I think it makes sense to restrict to parenthesized constraints. @aljoscha did a mini survey of DBMSes when he created the ticket, so he might have some thoughts. What do you think Aljoscha?

Reply:
When I originally wrote the proposal I added a `PRIMARY KEY ... NOT ENFORCED` without parentheses because the parentheses are used to specify the columns of the source. `(PRIMARY KEY ... NOT ENFORCED)` could be understood as specifying zero source columns and a primary-key constraint.

Reply:
That's a good point! I'd be inclined to keep it as is then, if @sploiselle agrees.

Reply:
Fine for me! I don't have a strong preference on this. And this keeps it simpler. 👍

Reply:
Changed! 👍🏼